fmutils.go

```diff
@@ -77,7 +77,16 @@ func (mask NestedMask) Filter(msg proto.Message) {
 				return true
 			}
 
-			if fd.IsList() {
+			if fd.IsMap() {
+				xmap := rft.Get(fd).Map()
+				xmap.Range(func(mk protoreflect.MapKey, _ protoreflect.Value) bool {
+					if _, ok := m[mk.String()]; !ok {
+						xmap.Clear(mk)
```

Comment:
If my understanding is correct this implementation does not support nested maps.
Would you be willing to add such a support, please?

For more context: consider the following proto field:

```protobuf
message M {
  map<string, map<string, string>> nested_tags = 1;
}
```

Reply:
Hmm, that's not valid protobuf syntax according to my understanding and the `protoc` compiler. Am I missing anything?

Reply:
```
testproto.proto:35:18: Expected ">".
```

Reply:
Hm... I did not know nested maps are not supported by protobuf, sorry.

But still there is a valid case when the value in the map is another protobuf message:

```protobuf
message M1 {
  map<string, M2> tags = 1;
}

message M2 {
  string first_name = 1;
  string last_name = 2;
}
```

Also keep in mind that recursive messages are possible:

```protobuf
message M1 {
  map<string, M2> tags = 1;
}

message M2 {
  string first_name = 1;
  string last_name = 2;
  map<int, M1> other_stuff = 3;
}
```

Reply:
Yes, that makes sense. Implemented that now.